pkg/proc/stack.go

```diff
@@ -543,6 +661,10 @@ func (it *stackIterator) advanceRegs() (callFrameRegs op.DwarfRegisters, ret uin
 		framectx = it.bi.Arch.FixFrameUnwindContext(fde.EstablishFrame(it.pc), it.pc, it.bi)
 	}
 
+	if _, ok := it.bi.Arch.(*ARM64); ok {
```

Comment:
I think this if and the call to switchSp can be moved outside of `advanceRegs` and directly into `Next`.

Reply:
The `switchSp` should be a part of `advanceRegs`, because it must get right SP value before get other regs. And the `advanceRegs()` called in two places of proc.go, that means `switchSp()` will appear twice too, that's duplicated i think.

Reply:
The other place that calls advanceRegs is amd64's switchStack for which it isn't relevant.

Reply:
I reorganize `switchSp` to `Arch interface`, is it ok?

Reply:
Yes, that's good, but it should still be called in Next instead of advanceRegs.